src/adapter-types/src/timestamp_selection.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use serde::{Deserialize, Serialize};
+
+/// Whether to use the constraint-based timestamp selection.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum ConstraintBasedTimestampSelection {
+    Enabled,
+    Disabled,
+    Verify,
+}
+
+impl std::default::Default for ConstraintBasedTimestampSelection {
+    fn default() -> Self {
+        Self::Verify
+    }
+}
+
+impl ConstraintBasedTimestampSelection {
+    pub const fn const_default() -> Self {
+        Self::Verify
+    }
+
+    pub fn from_str(s: &str) -> Self {
+        match s {
+            "enabled" => Self::Enabled,
+            "disabled" => Self::Disabled,
+            "verify" => Self::Verify,
+            _ => {
+                tracing::error!("invalid value for ConstraintBasedTimestampSelection: {}", s);
+                ConstraintBasedTimestampSelection::default()
+            }
+        }
+    }
```

Comment:
Could implement this as `TryFrom` instead, but totally up to you. It would have the benefit that the caller needs to worry about the error path, not the implementation.

Reply:
Dov originally used `TryFrom` but I recommended he push the error handling into this method so the caller doesn't have to worry about it :) In my experience the caller would generally just fallback to some default, unifying that across all callers seemed like a good idea